proof_of_play.go

```diff
@@ -87,21 +95,31 @@ func (p *proofOfPlay) start() {
 		if req.Status {
 			err := p.confirm(req.Ad, req.DisplayTime)
 			if err != nil {
-				p.publishEvent("ad-pop-failed",
-					fmt.Sprintf("adId: %s, error: %s", req.Ad["id"], err.Error()),
-					"critical")
+				p.processRequestFailure(req, err)
 			}
 		} else {
 			err := p.expire(req.Ad)
 			if err != nil {
-				p.publishEvent("ad-expire-failed",
-					fmt.Sprintf("adId: %s, error: %s", req.Ad["id"], err.Error()),
-					"critical")
+				p.processRequestFailure(req, err)
 			}
 		}
 	}
 }
 
+func (p *proofOfPlay) processRequestFailure(req *PoPRequest, err error) {
+	// Only raise the event on the first failed attempt
+	if req.RequestTime.IsZero() {
+		popType := p.getPoPType(req)
+		p.publishEvent(
+			fmt.Sprintf("ad-%s-failed", popType),
+			fmt.Sprintf("adId: %s, error: %s", req.Ad["id"], err.Error()),
+			"warning")
```

Comment:
this formatting looks very strange to me. did you run go fmt on this file?

Reply:
I just did and it looks OK. What looks strange about it?